File: app/src/main/java/com/sampietro/NaiveAAC/activities/Info/RequestConsentFirebaseActivity.java
package com.sampietro.NaiveAAC.activities.Info;

import android.content.Context;
import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.os.Handler;
import android.view.View;

import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentTransaction;

import com.google.firebase.analytics.FirebaseAnalytics;
import com.google.firebase.crashlytics.FirebaseCrashlytics;
import com.sampietro.NaiveAAC.R;
import com.sampietro.NaiveAAC.activities.Account.AccountActivity;
import com.sampietro.NaiveAAC.activities.Game.ChoiseOfGame.ChoiseOfGameActivity;
import com.sampietro.NaiveAAC.activities.Settings.Utils.AccountActivityAbstractClass;
import com.sampietro.NaiveAAC.activities.Settings.Utils.SettingsFragmentAbstractClass;

/**
 * <h1>RequestConsentFirebaseActivity</h1>
 * <p><b>RequestConsentFirebaseActivity</b> requires consent Firebase Analytics and Crashlytics</p>
 *
 * @version     3.0, 03/12/23
 * @see AccountActivityAbstractClass
 * @see SettingsFragmentAbstractClass
 */
public class RequestConsentFirebaseActivity extends AppCompatActivity {
    public static final String EXTRA_MESSAGE ="helloworldandroidMessage" ;
    //
    public View rootViewFragment;
    public Context context;
    public SharedPreferences sharedPref;
    //
    public FragmentManager fragmentManager;
    //
    /**
     * configurations of account start screen.
     *
     * @param savedInstanceState Define potentially saved parameters due to configurations changes.
     * @see RequestConsentFirebaseFragment
     * @see android.app.Activity#onCreate(Bundle)
     */
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_game);
        //
        context = this;
        sharedPref = context.getSharedPreferences(
                getString(R.string.preference_file_key), Context.MODE_PRIVATE);
        //
        if (savedInstanceState == null)
        {
            fragmentManager = getSupportFragmentManager();
            fragmentManager.beginTransaction()
                    .add(R.id.game_container, new RequestConsentFirebaseFragment(), "RequestConsentFirebaseFragment")
                    .commit();
        }
        //
    }
    /**
     * Called when the user taps the consent Firebase button.
     * </p>
     *
     * @param view view of tapped button
     */
    public void iAccept(View view) {
        //
        // consent firebase
        FirebaseAnalytics.getInstance(this).setAnalyticsCollectionEnabled(true);
        FirebaseCrashlytics.getInstance().setCrashlyticsCollectionEnabled(true);
        //
        boolean hasLastPlayer = verifyLastPlayer();
        if (hasLastPlayer) {
            // I move on to the activity of choice
            // Time to launch the another activity
            int TIME_OUT = 4000;
            new Handler().postDelayed(new Runnable() {
                @Override
                public void run() {

                    Intent i = new Intent(context,
                            ChoiseOfGameActivity.class);
                    //
                    String message = "Bentornato " ;
                    i.putExtra(EXTRA_MESSAGE, message);
                    startActivity(i);
                    finish();
                }
            }, TIME_OUT);
        }
    }
    /**
     * Called when the user taps the i do not consent Firebase button.
     * </p>
     *
     * @param view view of tapped button
     */
    public void iDoNotAccept(View view) {
        //
        boolean hasLastPlayer = verifyLastPlayer();
        if (hasLastPlayer) {
            // I move on to the activity of choice
            // Time to launch the another activity
            int TIME_OUT = 4000;
            new Handler().postDelayed(new Runnable() {
                @Override
                public void run() {

                    Intent i = new Intent(context,
                            ChoiseOfGameActivity.class);
                    //
                    String message = "Bentornato " ;
                    i.putExtra(EXTRA_MESSAGE, message);
                    startActivity(i);
                    finish();
                }
            }, TIME_OUT);
        }
    }
    //
    /**
     * if there is no player already registered, go to the user registration activity.
     *
     * @return boolean whit true if has last player
     * @see AccountActivity
     */
    public  boolean verifyLastPlayer() {
        boolean hasLastPlayer = sharedPref.contains(getString(R.string.preference_LastPlayer));
        if (!hasLastPlayer) {
            // go to the user registration activity
            Intent intent = new Intent(this, AccountActivity.class);
            //
            String message = "enter username";
            //
            intent.putExtra(EXTRA_MESSAGE, message);
            startActivity(intent);
            return false;
        }
        else
        {
            return true;
        }
    }
}

File: app/src/main/java/com/sampietro/NaiveAAC/activities/Info/RequestConsentFirebaseActivity.kt
package com.sampietro.NaiveAAC.activities.Info

import android.content.Context
import androidx.appcompat.app.AppCompatActivity
import android.content.SharedPreferences
import android.os.Bundle
import com.sampietro.NaiveAAC.R
import com.google.firebase.analytics.FirebaseAnalytics
import com.google.firebase.crashlytics.FirebaseCrashlytics
import android.content.Intent
import android.os.Handler
import android.os.Looper
import android.view.View
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import com.sampietro.NaiveAAC.activities.Game.ChoiseOfGame.ChoiseOfGameActivity
import com.sampietro.NaiveAAC.activities.Account.AccountActivity

/**
 * <h1>RequestConsentFirebaseActivity</h1>
 *
 * **RequestConsentFirebaseActivity** requires consent Firebase Analytics and Crashlytics
 *
 * @version     5.0, 01/04/2024
 */
class RequestConsentFirebaseActivity : AppCompatActivity() {
    //
//    var rootViewFragment: View? = null
    lateinit var context: Context
    lateinit var sharedPref: SharedPreferences

    //
    lateinit var fragmentManager: FragmentManager
    //
    /**
     * configurations of account start screen.
     *
     * @param savedInstanceState Define potentially saved parameters due to configurations changes.
     * @see RequestConsentFirebaseFragment
     *
     * @see android.app.Activity.onCreate
     */
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_game)
        //
        context = this
        sharedPref = getSharedPreferences(
            getString(R.string.preference_file_key), MODE_PRIVATE
        )
        //
        if (savedInstanceState == null) {
            fragmentManager = supportFragmentManager
            fragmentManager.beginTransaction()
                .add(
                    R.id.game_container,
                    Fragment(R.layout.activity_requestconsentfirebase),
                    "RequestConsentFirebaseFragment"
                )
                .commit()
        }
        //
    }

    /**
     * Called when the user taps the consent Firebase button.
     *
     *
     * @param view view of tapped button
     */
    fun iAccept(view: View?) {
        //
        // consent firebase
        FirebaseAnalytics.getInstance(this).setAnalyticsCollectionEnabled(true)
        FirebaseCrashlytics.getInstance().setCrashlyticsCollectionEnabled(true)
        //
        val hasLastPlayer = verifyLastPlayer()
        if (hasLastPlayer) {
            // I move on to the activity of choice
            // Time to launch the another activity
            val TIME_OUT = 4000
            Handler(Looper.getMainLooper()).postDelayed({
                val i = Intent(
                    context,
                    ChoiseOfGameActivity::class.java
                )
                //
                val message = getString(R.string.bentornato)
                i.putExtra(EXTRA_MESSAGE, message)
                startActivity(i)
                finish()
            }, TIME_OUT.toLong())
        }
    }

    /**
     * Called when the user taps the i do not consent Firebase button.
     *
     *
     * @param view view of tapped button
     */
    fun iDoNotAccept(view: View?) {
        //
        val hasLastPlayer = verifyLastPlayer()
        if (hasLastPlayer) {
            // I move on to the activity of choice
            // Time to launch the another activity
            val TIME_OUT = 4000
            Handler(Looper.getMainLooper()).postDelayed({
                val i = Intent(
                    context,
                    ChoiseOfGameActivity::class.java
                )
                //
                val message = getString(R.string.bentornato)
                i.putExtra(EXTRA_MESSAGE, message)
                startActivity(i)
                finish()
            }, TIME_OUT.toLong())
        }
    }
    //
    /**
     * if there is no player already registered, go to the user registration activity.
     *
     * @return boolean whit true if has last player
     * @see AccountActivity
     */
    fun verifyLastPlayer(): Boolean {
        val hasLastPlayer = sharedPref.contains(getString(R.string.preference_LastPlayer))
        return if (!hasLastPlayer) {
            // go to the user registration activity
            val intent = Intent(this, AccountActivity::class.java)
            //
            val message = "enter username"
            //
            intent.putExtra(EXTRA_MESSAGE, message)
            startActivity(intent)
            false
        } else {
            true
        }
    }

    companion object {
        const val EXTRA_MESSAGE = "helloworldandroidMessage"
    }
}